grpc/diatheke.proto

```diff
@@ -46,13 +46,36 @@ service Diatheke {
     // endpoint), or when a transcript becomes available on its
     // own, in which case the stream is closed by the server.
     // The ASR result may be used in the UpdateSession method.
+    //
+    // If the session has a wakeword enabled, and the client
+    // application is using Diatheke and Cubic to handle the
+    // wakeword processing, this method will not return a
+    // result until the wakeword condition has been satisfied.
+    // Utterances without the required wakeword will be
+    // discarded and no transcription will be returned.
     rpc StreamASR(stream ASRInput) returns (ASRResult) {}
 
     // Create a TTS stream to receive audio for the given reply.
     // The stream will close when TTS is finished. The client
     // may also close the stream early to cancel the speech
     // synthesis.
     rpc StreamTTS(ReplyAction) returns (stream TTSAudio) {}
+
+    // Create an ASR stream for transcription. Unlike StreamASR,
+    // Transcribe does not listen for a wakeword. This method
+    // returns a bi-directional stream, and its intended use is
+    // for situations where a user may say anything at all, whether
+    // it is short or long, and the application wants to save the
+    // transcript (e.g., take a note, send a message).
+    //
+    // The first message sent to the server must include the
+    // Cubic model ID, with remaining messages sending audio data.
+    // Messages received from the server will include the current
+    // best partial transcription until the full transcription is
+    // ready. The stream ends when either the client application
+    // closes it, a predefined duration of silence (non-speech)
```

Comment:
Does "predefined" mean in the Diatheke config?  Also, is the end-transcription intent one that has a specific kind of action that Diatheke handles rather than returning it to the client, or is it a normal command action that the client application should handle by closing the stream?

Reply:
"predefined" means on the server side. It is more likely that this will be in the model rather than the server config.

> is the end-transcription intent one that has a specific kind of action that Diatheke handles rather than returning it to the client

Yes. The idea is to specify the intent in the model, and Diatheke will listen for that intent and automatically close the stream on the server side.